visualization/app/codeCharta/ui/codeMap/codeMap.mouseEvent.service.ts

```diff
@@ -186,24 +189,27 @@ export class CodeMapMouseEventService
 	}
 
 	private hoverBuilding(hoveredBuilding: CodeMapBuilding) {
-		/*
-         if the hovered node does not have useful data, then we should look at its parent. If the parent has useful data
-         then this parent is a delta node which is made of two seperate, data-free nodes. This quick fix helps us to
-         handle delta objects, until there is a method for mergng their meshes and materials correctly.
-         See codeMapRenderService.js
-         */
-		if (hoveredBuilding && !hoveredBuilding.node) {
-			if (hoveredBuilding.parent && hoveredBuilding.parent.node) {
-				hoveredBuilding.setNode(hoveredBuilding.parent.node)
-			}
-		}
-
 		if (hoveredBuilding) {
-			this.threeSceneService.highlightSingleBuilding(hoveredBuilding)
-			this.$rootScope.$broadcast(CodeMapMouseEventService.BUILDING_HOVERED_EVENT, { hoveredBuilding: hoveredBuilding })
+			this.hoverBuildingAndChildren(hoveredBuilding)
 		}
 	}
 
+	private hoverBuildingAndChildren(hoveredBuilding: CodeMapBuilding) {
+		const lookUp = this.storeService.getState().lookUp
+		const codeMapNode = lookUp.pathToNode.get(hoveredBuilding.node.path)
+		this.threeSceneService.addBuildingToHighlightingList(hoveredBuilding)
+		hierarchy(codeMapNode)
+			.descendants()
+			.forEach(x => {
+				const building = lookUp.pathToBuilding.get(x.data.path)
+				if (building) {
+					this.threeSceneService.addBuildingToHighlightingList(building)
+				}
+			})
```

Comment:
Why not just use the `x.data.node` without using the lookUp?

Reply:
Okey never mind. Its because we can not access the children on the codeMapBuilding class, right?

Reply:
x.data.node is of type Node, not CodeMapNode.